src/visualizers/widgets/MonacoEditor/MonacoEditorWidget.js

```diff
@@ -0,0 +1,476 @@
+        this.modelURI = modelURI + chance.string({length: 5}) + extension;
+        this.monacoURI = monaco.Uri.parse(this.modelURI);
+        this.language = config.language || 'python';
+        this.keyBindingAdapter = null;
+
+        this._el = container;
+        this._el.css({height: '100%'});
+        this._el.append($(TAB_NAV));
+
+        this.$editor = $('<div/>');
+        this.$status = $('<div/>');
+        this.$editor.css({height: '100%'});
+        this._el.append(this.$editor[0]);
+        this._el.append(this.$status[0]);
+
+        this.readOnly = config.readOnly || false;
+
+        // register context menu to display settings and ComponentSettings
+        // Currently there's no submenu in monaco-editor and context menu is used
+        // to add display settings. See https://github.com/microsoft/monaco-editor/issues/1947
+        this._registerContextMenu();
+
+        // register languages to the monaco editor
+        this._registerLanguages();
+
+        // Create editor with value provided by constructor
+        const value = config.value || "def dummy_python_func():\n\tpass";
+        this.model = monaco.editor.getModel(this.monacoURI) ||
+            monaco.editor.createModel(
+                value,
+                this.language,
+                this.monacoURI
+            );
+        this.model.updateOptions({
+            tabSize: 4,
+            insertSpaces: true
+        });
+
+        const dontDisplayMiniMap = config.dontDisplayMiniMap || false;
+        this.editor = this._createEditor(dontDisplayMiniMap);
+        this.DELAY = 750;
+        this.silent = false;
+        this.saving = false;
+        this.count = 0;
+
+        this.model.onDidChangeContent(() => {
+            if (!this.silent) {
+                this.saving = true;
+                this.onChange();
+            }
+        });
+
+        this.onChange = _.debounce(this.saveText.bind(this), this.DELAY);
+        this.currentHeader = '';
+        this.activeNode = null;
+
+        this.$settingsBtn = this._el.find(`.${DISPLAY_MENU_CLASS}`);
+
+        this.nodes = {};
+        this._initialize();
+        this._logger.debug('ctor finished');
+    }
+
+    MonacoEditorWidget.prototype._createEditor = function (dontDisplayMiniMap) {
+        if (!DEFAULT_THEMES.includes(this.displaySettings.theme)) {
+            this.importTheme(this.displaySettings.theme, ThemeList[this.displaySettings.theme]);
+            this.setNavColor(this.displaySettings.navColor);
+        }
+        const editor = monaco.editor.create(
+            this.$editor[0],
+            {
+                model: this.model,
+                automaticLayout: true,
+                lightbulb: {
+                    enabled: true
+                },
+                fontSize: this.displaySettings.fontSize,
+                readOnly: this.readOnly,
+                minimap: {
+                    enabled: !dontDisplayMiniMap
```

Comment:
See my earlier comment about `dontDisplayMinimap` - this would be easier to understand if it was just `displayMinimap` rather than `!dontDisplayMinimap`.